mindmeld/components/entity_resolver.py

```diff
@@ -538,12 +618,101 @@ def _construct_whitelist_query(entity, weight=1, use_phons=False):
+                e.error,
+                e.status_code,
+                e.info,
+            )
+            raise EntityResolverError from e
+        except ElasticsearchException as e:
+            raise EntityResolverError from e
+
+
+class EntityResolverUsingExactMatch(EntityResolverBase):
+    """
+    Resolver class based on exact matching
+    """
+
+    def __init__(self, app_path, resource_loader, entity_type, er_config, **kwargs):
+        super().__init__(app_path, resource_loader, entity_type, er_config, **kwargs)
+        self._normalizer = self.resource_loader.query_factory.normalize
+        self._exact_match_mapping = None
+
+    @staticmethod
+    def _process_entity_map(entity_type, entity_map, normalizer):
+        """Loads in the mapping.json file and stores the synonym mappings in a item_map and a
+        synonym_map for exact match entity resolution when Elasticsearch is unavailable
 
         Args:
-            entity (Entity): An entity found in an input query
+            entity_type: The entity type associated with this entity resolver
+            entity_map: The loaded mapping.json file for the given entity type
+            normalizer: The normalizer to use
+        """
+        item_map = {}
+        syn_map = {}
+        seen_ids = []
+        for item in entity_map.get("entities"):
+            cname = item["cname"]
+            item_id = item.get("id")
+            if cname in item_map:
+                msg = "Canonical name %s specified in %s entity map multiple times"
+                logger.debug(msg, cname, entity_type)
+            if item_id:
+                if item_id in seen_ids:
+                    msg = "Item id {!r} specified in {!r} entity map multiple times"
+                    raise ValueError(msg.format(item_id, entity_type))
+                seen_ids.append(item_id)
+
+            aliases = [cname] + item.pop("whitelist", [])
+            items_for_cname = item_map.get(cname, [])
+            items_for_cname.append(item)
+            item_map[cname] = items_for_cname
+            for alias in aliases:
+                norm_alias = normalizer(alias)
+                if norm_alias in syn_map:
+                    msg = "Synonym %s specified in %s entity map multiple times"
+                    logger.debug(msg, cname, entity_type)
+                cnames_for_syn = syn_map.get(norm_alias, [])
+                cnames_for_syn.append(cname)
+                syn_map[norm_alias] = list(set(cnames_for_syn))
+
+        return {"items": item_map, "synonyms": syn_map}
+
+    def _fit(self, clean):
+        """Loads an entity mapping file to resolve entities using exact match.
         """
+        if clean:
+            logger.info(
+                "clean=True ignored while fitting exact_match algo for entity resolution"
+            )
+
+        entity_map = self.resource_loader.get_entity_map(self.type)
+        self._exact_match_mapping = self._process_entity_map(
```

Comment:
Since the args to this function are set once, if we already have `_exact_match_mapping ` populated, then cant we just cache this attribute and not call this function the second time?